src/Board.cpp

```diff
@@ -0,0 +1,211 @@
+#pragma once
+
+#include "Board.h"
+
+/**
+ * Get the singleton instance of the Board class.
+ *
+ * @return The Board instance.
+ */
+Board& Board::getInstance() {
+    static Board instance;
+    return instance;
+}
+
+/**
+ * Move a piece on the chess board.
+ *
+ * @param moveString The move to be made in string format.
+ * @return An integer representing the result of the move.
+ *         11: The starting position is not occupied by any piece.
+ *         12: The player is trying to move a piece that is not theirs.
+ *         13: The player is trying to capture their own piece.
+ *         21: The move is not valid for the piece.
+ *         31: The move causes the current player to check.
+ *         41: The move results in the current player being in check.
+ *         42: The move is successful and the turn is switched.
+ */
+int Board::move(const std::string& moveString) {
+    auto moveCoords = parseMoveString(moveString);
+    auto& startPos = moveCoords.first;
+    auto& endPos = moveCoords.second;
+    auto& currentPiece = m_chessMatrix[startPos.first][startPos.second];
+
+    if (!currentPiece) {
+        return 11;
+    }
+
+    if (currentPiece->getColor() != m_currentPlayer) {
+        return 12;
+    }
+
+    if (m_chessMatrix[endPos.first][endPos.second] &&
+        m_chessMatrix[endPos.first][endPos.second]->getColor() == m_currentPlayer) {
+        return 13;
+    }
+
+    if (!currentPiece->isValidMove(moveCoords, m_chessMatrix)) {
+        return 21;
+    }
+
+    if (isCheck(m_currentPlayer)) {
+        m_chessMatrix[startPos.first][startPos.second] = std::move(m_chessMatrix[endPos.first][endPos.second]);
+        return 31;
+    }
+
+    makeMove(moveCoords);
+
+    if (isCheck(m_currentPlayer)) {
+        return 41;
+    }
+
+    return 42;
+}
+
+/**
+ * Parse the move string into coordinates.
+ *
+ * @param moveString The move to be parsed.
+ * @return A pair of pairs, where the first pair is the starting position
+ *         and the second pair is the ending position.
+ */
+std::pair<std::pair<int, int>, std::pair<int, int>> Board::parseMoveString(const std::string& moveString) const {
+    std::pair<int, int> startPos = { -1, -1 };
+    std::pair<int, int> endPos = { -1, -1 };
+    int row = moveString[0] - 'a';
```

Comment:
you can remove the repetitive code for extracting row and col values. You can consider writing a helper function that takes a character and returns the corresponding row or column value.

Reply:
Done.